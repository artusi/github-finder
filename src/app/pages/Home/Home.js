import React, { Component } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import PropTypes from "prop-types";
import Header from "components/Header";
import { requestUserUpdate } from "store/state/github";
import { SearchField } from "ui";
import styles from "./Home.module.css";

export class Home extends Component {
  componentDidMount() {
    this.props.requestUserUpdate();
  }

  render() {
    return (
      <div className={styles.home}>
        <Header />
        <div className={styles.searchWrapper}>
          <div className={styles.searchUser}>
            <SearchField
              id="searchUser"
              onClick={() => {}}
              prefix="https://github.com/"
              label="Search a GitHub user:"
            />
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  github: state.github
});

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      requestUserUpdate
    },
    dispatch
  );
}

Home.propTypes = {
  requestUserUpdate: PropTypes.func.isRequired
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Home);
