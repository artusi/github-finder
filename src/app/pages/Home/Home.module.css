.home {
}
.searchWrapper {
  text-align: center;
  margin: 0 auto;
}
.searchUser {
  padding: 16px;
  border-radius: 12px;
  background: #f2cd5d;
  max-width: 420px;
}
