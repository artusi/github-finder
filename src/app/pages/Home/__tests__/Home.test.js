import React from "react";
import { Home } from "../Home";

describe("<Home />", () => {
  it("renders", () => {
    const wrapper = shallow(<Home match={{ params: {} }} />);

    expect(wrapper).toMatchSnapshot();
  });
});
