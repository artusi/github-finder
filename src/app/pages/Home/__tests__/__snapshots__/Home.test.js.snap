// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders 1`] = `
<div
  className="home"
>
  <Header />
  <Connect(FindUser)
    initialUser="artusi"
    onUpdateUser={[Function]}
  />
  <div
    className="reposWrapper"
  >
    <Connect(RepositoriesList) />
  </div>
</div>
`;
